File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
         long n = nums.length;
       Arrays.sort(nums);
        int l =0;  long sum =0;
        for(int i=0;i<n;i++){
             long target = nums[i];
            sum += target;
            if((i-l+1)*target-sum>k){
                sum-= nums[l];
                l++;
            }
        }
        return nums.length -l;
    }
}

File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        long long n = nums.size();
        sort(nums.begin(), nums.end());
        int l =0; long long sum =0;
        for(int i=0;i<n;i++){
            long long target = nums[i];
            sum += target;
            if((i-l+1)*target-sum>k){
                sum-= nums[l];
                l++;
            }
        }
        return n-l;
    }
};